src/Compilers/CSharp/Portable/Emitter/NoPia/EmbeddedTypesManager.cs

```diff
@@ -233,10 +233,18 @@ internal override void ReportIndirectReferencesToLinkedAssemblies(AssemblySymbol
                 case TypeKind.Interface:
                     foreach (Symbol member in namedType.GetMembersUnordered())
                     {
-                        if (member.Kind != SymbolKind.NamedType && !member.IsAbstract)
+                        if (member.Kind != SymbolKind.NamedType)
                         {
-                            error = ErrorCode.ERR_DefaultInterfaceImplementationInNoPIAType;
-                            break;
+                            if (!member.IsAbstract)
+                            {
+                                error = ErrorCode.ERR_DefaultInterfaceImplementationInNoPIAType;
+                                break;
```

Comment:
This does not feel like the right behavior for a constant declared in an interface.  Similarly for a static method.  I'm not sure we intend to forbid them in an embedded interface, but even if we do this seems like the wrong diagnostic.

Reply:
> This does not feel like the right behavior for a constant declared in an interface.  Similarly for a static method.  

NoPia feature is not supposed to embed constants or static methods. BTW, constants are never referenced explicitly in metadata, therefore they do not come through this code path.

> I'm not sure we intend to forbid them in an embedded interface, but even if we do this seems like the wrong diagnostic.

Could you please elaborate what is wrong with it?

---
In reply to: [315919836](https://github.com/dotnet/roslyn/pull/38140#discussion_r315919836) [](ancestors = 315919836)